migrations/versions/0321_update_postage_constraint_1.py

```diff
@@ -0,0 +1,85 @@
+"""
+
+Revision ID: 0321_update_postage_constraint_1
+Revises: 0320_optimise_notifications
+Create Date: 2020-05-12 16:17:21.874281
+
+"""
+import os
+
+from alembic import op
+
+
+revision = '0321_update_postage_constraint_1'
+down_revision = '0320_optimise_notifications'
+environment = os.environ['NOTIFY_ENVIRONMENT']
+
+
+def upgrade():
+    op.drop_constraint('chk_notifications_postage_null', 'notifications')
+    op.execute("""
+        ALTER TABLE notifications ADD CONSTRAINT "chk_notifications_postage_null"
+        CHECK (
+            CASE WHEN notification_type = 'letter' THEN
+                postage is not null and postage in ('first', 'second', 'europe', 'rest-of-world')
+            ELSE
+                postage is null
+            END
+        )
+        NOT VALID
+    """)
+
+    # The constraint on notification_history will be dropped manually
+    if environment not in ["live", "production"]:
+        op.drop_constraint('chk_notification_history_postage_null', 'notification_history')
```

Comment:
I had to comment out this drop constraint on my local, too. Probably because I had this upgrade installed before, and then had some trouble downgrading, and was deleting constraints manually.

So for preview and staging it should be fine, because downgrading went smoother there when we were reverting this PR or something?

Reply:
For preview and staging we need to run this step because the constraint still exists. In production the constraint has already been dropped manually, so this line would give an error if we tried to run it there.

Reply:
We can change the constraint to say `ALTER TABLE notification_history DROP CONSTRAINT IF EXISTSchk_notifications_postage_null` Then we don't need the `if environment` check.

Reply:
👍 That's updated now @servingUpAces